addons/mail/models/mail_rotting_resource_mixin.py

```diff
@@ -0,0 +1,55 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+from datetime import timedelta
+
+
+class RottingResourceMixin(models.AbstractModel):
+    _name = 'mail.rotting.resource.mixin'
+    _description = 'Mixin for resources that can rot. Needs to be related to a mail.rotting.stage.mixin'
+
+    _inherit = ['mail.thread']
+
+    date_rot = fields.Date('Last activity', compute="_compute_date_rot", store=True)
+    is_rotting = fields.Boolean('Rotting', compute='_compute_rotting')
+    day_rotting = fields.Integer('Days Rotting', help='Day count since this resource was last updated',
+        compute='_compute_rotting')
+    # existence check: the parent model needs a stage_id (inheriting from the stage mixin?)
+    stage_id = fields.Many2one('mail.rotting.stage.mixin')
```

Comment:
Probably to remove (we don't link mixin models). Everything should be covered by the helper giving the stage field (see duration mixin).